providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -244,11 +244,35 @@ static const OSSL_PARAM *rsa_importkey_types(void)
     return rsa_key_types;
 }
 
+static int rsa_validatekey_public(void *key)
```

Comment:
Done forget to add forward declarations at the top of this file, so the compiler can help out confirming that you got the signature right...